src/OpenTelemetry.AutoInstrumentation.ClrProfiler.Managed/Configuration/EnvironmentConfigurationHelper.cs

```diff
@@ -1,20 +1,38 @@
 using System;
+using System.Collections.Generic;
+using System.Linq;
 using OpenTelemetry.Resources;
 using OpenTelemetry.Trace;
 
 namespace OpenTelemetry.AutoInstrumentation.ClrProfiler.Managed.Configuration
 {
     internal static class EnvironmentConfigurationHelper
     {
+        private static readonly Dictionary<InstrumentationType, Action<TracerProviderBuilder>> AddInstrumentation = new()
+        {
+            [InstrumentationType.HttpClient] = builder => builder.AddHttpClientInstrumentation(),
+            [InstrumentationType.AspNet] = builder => builder.AddSdkAspNetInstrumentation(),
+            [InstrumentationType.SqlClient] = builder => builder.AddSqlClientInstrumentation()
+        };
+
         public static TracerProviderBuilder UseEnvironmentVariables(this TracerProviderBuilder builder, Settings settings)
         {
             var resourceBuilder = ResourceBuilder
                 .CreateDefault()
                 .AddService(settings.ServiceName ?? "UNKNOWN_SERVICE_NAME", serviceVersion: settings.ServiceVersion);
 
-            return builder
+            builder
                 .SetResourceBuilder(resourceBuilder)
                 .SetExporter(settings);
+
+            foreach (var enabledInstrumentation in settings.EnabledInstrumentations)
+            {
+                AddInstrumentation[enabledInstrumentation](builder);
+            }
+
+            builder.AddSource(settings.ActivitySources.ToArray());
```

Comment:
I think we should add the same for `AddLegacySource`
can be in a separate PR

Reply:
I will add it to this one